### DISCUSSION TITLE
Check DELPHI(LEP) exclusion using MA5

### DISCUSSION THREAD

**Discussion by arnab098** [MadAnalysis/madanalysis5#123] [Q&A] [answered]
Hi,
I am trying to check some DELPHI(LEP) exclusion for a model in the mono-photon+MET channel. I want to use MA5 for that purpose. How can I implement the detector response in this case? Is it good to add a detector card in Delphes, or somehow skip the Delphes simulation and put all efficiencies and smearing in a recast code? Or any other better way?

With best regards,
Arnab.

**Comment by jackaraz** [comment, marked as answer]
Hi Arnab, there are two internal detector simulators within MadAnalysis. One is Delphes; if you are familiar with it, you can implement detector response with a TLC card, and MadAnalysis will be able to read and generate simulated output for you. Then you can implement your analysis by using the output. The second option is [SFS](https://arxiv.org/abs/2006.09387), a native simulator to MadAnalysis 5. With SFS, you can either generate simulated data and analyze it or do the detector simulation and your analysis simultaneously where it applies the detector effects on the fly.

For either option, you have some default cards in [this folder](https://github.com/MadAnalysis/madanalysis5/tree/main/madanalysis/input) where the `.ma5` extension stands for SFS cards. Also, you can install `PAD` or `PADForSFS` and check the current implementations for recasting purposes. Validation notes can be found at [this link](http://madanalysis.irmp.ucl.ac.be/wiki/PublicAnalysisDatabase).

If none of them is at your speed, you can also implement your detector simulation in expert mode.

Cheers
Jack

**Reply by arnab098** [reply]
Thanks a lot, Jack. This will help.

With best regards,
Arnab.
